src/tree/updater_quantile_hist.cc

```diff
@@ -170,6 +173,8 @@ void QuantileHistMaker::Builder::BuildNodeStats(
       auto parent_split_feature_id = snode_[parent_id].best.SplitIndex();
       spliteval_->AddSplit(parent_id, left_sibling_id, nid, parent_split_feature_id,
```

Comment:
@hcho3 I'm not entirely sure why there are two splits on the pipeline.

Reply:
Interesting, I have no idea either. Let's dig into git blame.

Reply:
@hcho3 Seems to be #4102 .